radicle/src/cob/patch.rs

```diff
@@ -216,6 +216,15 @@ impl Patch {
             .author
     }
 
+    /// Get the `Revision` by its `RevisionId`.
+    ///
+    /// None is returned if the `Revision` has been redacted (deleted).
+    pub fn revision(&self, id: &RevisionId) -> Option<&Revision> {
+        self.revisions()
+            .find(|(id_, _)| id_ == &id)
+            .map(|(_, rev)| rev)
+    }
```

Comment:
The reason for adding this function was to have O(1) access to any revision; but this implementation just iterates over all revisions instead of just getting the revision from the map.